docs/syntax/types.md

```diff
@@ -276,32 +276,48 @@ context-dependent manner that is discussed properly in the
 [type system design document](../types/README.md), but is summarised briefly
 below.
 
-- **Name and Fields:** When you provide the keyword with only a name and some
-  field names, this creates an atom.
-
-  ```ruby
-  type Just value
-  ```
-
 - **Body with Atom Definitions:** If you provide a body with atom definitions,
   this defines a smart constructor that defines the atoms and related functions
   by returning a typeset.
 
   ```ruby
-  type Maybe a
+  type Maybe
       Nothing
-      type Just (value : a)
+      Just (value : Integer)
+
+      is_just self = case self of
+          Maybe.Nothing -> False
+          Maybe.Just _ -> True
+
+      nothing self = self.is_just.not
+  ```
 
-      isJust = case this of
-          Nothing -> False
-          Just _ -> True
+  To reference atoms use type name followed by the name of the atom. E.g.
+  `Maybe.Nothing` or `Maybe.Just 2`. Atom constructors act like functions and
+  fully support currying - e.g. one can create `fn = Maybe.Just` and later apply
+  two to it (`fn 2`) to obtain new atom.
 
-      nothing = not isJust
+- **Autoscoped Constructors:** Referencing constructors via their type name may
```

Comment:
Documenting _autoscoped constructors_ is hard when most of the surrounding documentation is outdated...